jffs2/libjffs2.c

```diff
@@ -530,7 +550,10 @@ static int jffs2_srv_create(jffs2_partition_t *p, oid_t *dir, const char *name,
 	dentry = malloc(sizeof(struct dentry));
 	memset(dentry, 0, sizeof(struct dentry));
 	dentry->d_name.name = strdup(name);
-	dentry->d_name.len = strlen(name);
+	if (dentry->d_name.name == NULL)
+		return -ENOMEM;
```

Comment:
Memory leak: `dentry`.
```suggestion
	if (dentry->d_name.name == NULL) {
		free(dentry);
		return -ENOMEM;
	}
```

Reply:
Thanks, done.